lambda/src/main/scala/pricemigrationengine/migrations/Newspaper2024Migration/Newspaper2024MigrationEstimation.scala

```diff
@@ -0,0 +1,253 @@
+                if (monthIndex <= 20) {
+                  Right(MonthliesPart2)
+                } else {
+                  Right(MonthliesPart1)
+                }
+              }
+            }
+          }
+          case _ => Right(MoreThanMonthlies)
+        }
+      }
+    }
+  }
+
+  def batchIdToEarliestMigrationStartDate(batchId: Newspaper2024BatchId): LocalDate = {
+    batchId match {
+      case MonthliesPart1    => LocalDate.of(2024, 2, 21) // 21 Feb 2024
+      case MonthliesPart2    => LocalDate.of(2024, 3, 18) // 18 March 2024
+      case MoreThanMonthlies => LocalDate.of(2024, 3, 1) // 1 March 2024
+    }
+  }
+
+  def subscriptionToEarliestMigrationStartDate(subscription: ZuoraSubscription): LocalDate = {
+    subscriptionToBatchId(subscription) match {
+      case Right(bid)   => batchIdToEarliestMigrationStartDate(bid)
+      case Left(string) => LocalDate.of(2024, 4, 1)
+      // Default date to avoid returning a more complex value than a LocalDate
+    }
+  }
+
+  def startDateGeneralLowerbound(
+      cohortSpec: CohortSpec,
+      today: LocalDate,
+      subscription: ZuoraSubscription
+  ): LocalDate = {
+
+    // Technically the startDateGeneralLowerbound is a function of the cohort spec and the notification min time.
+    // The cohort spec carries the lowest date we specify there can be a price migration, and the notification min
+    // time ensures the legally required lead time for customer communication. The max of those two dates is the date
+    // from which we can realistically perform a price increase. With that said, other policies can apply, for
+    // instance:
+    // - The one year policy, which demand that we do not price rise customers during the subscription first year
+    // - The spread: a mechanism, used for monthlies, by which we do not let a large number of monthlies migrate
+    //   during a single month.
+
+    // We expanded the signature of this function for the Newspaper2024 migration where that date was specific of
+    // the subscription due to its un-unusual scheduling. For Newspaper2024 we call a specific function from the
+    // migration support code.
+
+    val earliestPriceMigrationStartDate = subscriptionToEarliestMigrationStartDate(subscription)
+
+    def datesMax(date1: LocalDate, date2: LocalDate): LocalDate = if (date1.isBefore(date2)) date2 else date1
+
+    datesMax(
+      earliestPriceMigrationStartDate,
+      today.plusDays(
+        Newspaper2024MigrationStaticData.minLeadTime + 1
+      ) // +1 because we need to be strictly over minLeadTime days away. Exactly minLeadTime is not enough.
+    )
+  }
+
+  def startDateSpreadPeriod(subscription: ZuoraSubscription): Int = {
+    subscriptionToBatchId(subscription) match {
+      case Left(_) => 1
+      case Right(bid) =>
+        bid match {
+          case MonthliesPart1    => 1
+          case MonthliesPart2    => 2
+          case MoreThanMonthlies => 1
+        }
```

Comment:
Not sure if it's any clearer but could be:
```suggestion
      case Right(MonthliesPart1)    => 1
          case Right(MonthliesPart2)    => 2
          case Right(MoreThanMonthlies) => 1
        }
```

Reply:
Nice one. It's now
```
    subscriptionToBatchId(subscription) match {
      case Right(MonthliesPart1)    => 1
      case Right(MonthliesPart2)    => 2
      case Right(MoreThanMonthlies) => 1
      case Left(_)                  => 1
    }
```